students/piotrowski_stanislaw/lesson_03_functions/collatz_input_validation.py

```diff
@@ -0,0 +1,30 @@
+# https://automatetheboringstuff.com/chapter3/
+# "The Collatz Sequence + Input Validation"
+# piotrsta
+
+
+def collatz(argument):
+    try:
```

Comment:
łapanie wyjątku w tym samym miejscu co rzucanie, nie jest najlepszym podejściem.
Wyjątek wewnątrz funkcji rzucasz, żeby przekazac na zewnątrz/ w miejsce wywołania funkcji, informację o błędzie.
Tak więc blok `try:.. catch` powinieneś nastepnym razem użyć w miejscu gdzie wywołałeś ta funkcję.
Możesz również przy rzucaniu wyjątku podać wiadomość, dlaczego go rzuciłeś i odczytać przy łapaniu:
`raise ValueError`
i złapać:
`except ValueError as e:`
`    print(e)`
Dzieki temu podejściu w funkcji możesz zdefniowac wiele przypadków rzucenia tego wyjątku i dla każdego zdefniować osobną wiadomość. A przy łapaniu odrazu wiesz dlaczego to się stało.